packages/themes/src/createTheme.ts

```diff
@@ -1,20 +1,317 @@
+  }
+
+  let cssString: string | null = null;
+  if (lightRules.length > 0 && darkModeSelector !== null) {
+    const normalizedSelector = darkModeSelector as string;
+
+    if (normalizedSelector.startsWith(MEDIA_QUERY_PREFIX)) {
+      // Media query - wrap dark rules in :root
+      cssString = `
+    :root {
+${lightRules.join('\n')}
+    }
+    ${normalizedSelector} {
+      :root {
+${darkRules.join('\n')}
+      }
+    }
+  `.trim();
+    } else {
+      // Regular selector - use as-is
+      cssString = `
+    :root {
+${lightRules.join('\n')}
+    }
+    ${normalizedSelector} {
+${darkRules.join('\n')}
+    }
+  `.trim();
+    }
+  }
+
+  return { cssString, processedVariables };
+}
+
+/**
+ * Transforms a CSS selector for proper nesting context
+ * Class selectors like '.dark' become '.dark &' for parent-child relationship
+ * ID selectors like '#theme' become '#theme &' for parent-child relationship
+ */
+export function transformSelectorForNesting(selector: string): string {
+  // If it's a class or ID selector, append & for proper CSS nesting
+  if (
+    (selector.startsWith(CLASS_SELECTOR_PREFIX) || selector.startsWith(ID_SELECTOR_PREFIX)) &&
+    !selector.includes('&')
+  ) {
+    return `${selector} &`;
+  }
+  // For media queries and other selectors, return as-is
+  return selector;
+}
+
+/**
+ * Resolves elements configuration - handles both static elements and function-based elements
+ */
+export function resolveElementsConfiguration(
+  elements: Elements | DarkModeElementsFunction | undefined,
+  darkModeSelector: string | null,
+): Elements | undefined {
+  if (!elements) {
+    return undefined;
+  }
+
+  // If it's a function, call it with darkModeSelector
+  if (typeof elements === 'function') {
+    // If opted out of dark mode, don't call the function at all
+    // since it's designed to generate dark mode styles
+    if (darkModeSelector === null) {
+      return undefined;
+    }
+
+    try {
+      // Use the explicit selector provided (no fallbacks)
+      const rawSelector = darkModeSelector;
+      const selector = transformSelectorForNesting(rawSelector);
+      return (elements as DarkModeElementsFunction)(selector);
+    } catch (error) {
+      console.warn('[Clerk Theme] Failed to call elements function with darkModeSelector:', error);
+      return undefined;
+    }
+  }
+
+  return elements;
+}
+
+/**
+ * Resolves a baseTheme, handling both static themes and factory functions
+ */
+export function resolveBaseTheme(baseTheme: BaseTheme | undefined, options?: ThemeOptions): BaseTheme | undefined {
+  if (!baseTheme) {
+    return undefined;
+  }
+
+  // If baseTheme is a factory function, call it with the same options
+  if (typeof baseTheme === 'function') {
+    try {
+      return (baseTheme as any)(options);
+    } catch (error) {
+      console.warn('[Clerk Theme] Failed to call baseTheme factory function:', error);
+      return baseTheme as BaseTheme; // Fallback to treating it as static theme
```

Comment:
`baseTheme` is a function in this case right? This return seems problematic, we're casting a function as `BaseTheme`